### PR TITLE
Environment variables to configure custom headless server & macOS build fix

### PR DESCRIPTION
**Pull request by dahlia**
- Defined new environment variables `NC_RPC_SERVER_HOST`, `NC_RPC_SERVER_PORT`, `NC_GRAPHQL_SERVER_HOST`, and `NC_GRAPHQL_SERVER_PORT` to configure custom headless server.
- Fixed broken build on macOS.

### CODE REVIEW

**Review by boscohyun — APPROVED** [review]

**Review by moreal — COMMENTED** [review]
It seems there is no dependencies changes, but there is the change of *package-lock.json*. Could you check it if you missed to add *package.json*?

**Comment by dahlia** [comment]
> It seems there is no dependencies changes, but there is the change of _package-lock.json_. Could you check it if you missed to add _package.json_?

Missed nothing, but it was changed after I simply ran `npm install --dev`.

**Comment by dahlia** [comment]
Rebased on development (was main).

File: src/config.ts [hunk 1]
```diff
@@ -103,21 +103,45 @@ export const electronStore = new Store<IElectronStore>({
...
+
   // FIXME: 열려 있지 않는 랜덤한 포트를 반환하게 해야 합니다.
-  return 23142;
+  return { port: 23142, notDefault: false };
 };
 
-const LocalServerPort = (): number => {
+const LocalServerHost = (): { host: string; notDefault: boolean } => {
+  const host = process.env.NC_GRAPHQL_SERVER_HOST;
+  return host == null
```
Comment (moreal) [inline src/config.ts:132]: This comparison will be `true` when also `host` is undefined. Is it intended? If not, how about using `===` and `!==` operator.

**Review by moreal — APPROVED** [review]

File: src/config.ts (hunk 1 above)
Comment (dahlia) [reply src/config.ts:132]: That's intended. 😎

**Review by ipdae — APPROVED** [review]

**Comment by ipdae** [comment]
conflict on main.ts

**Comment by dahlia** [comment]
@riemannulus Resolved the conflict.

**Review by riemannulus — APPROVED** [review]

**Pull request merged**